### PR TITLE
Well connections prep for conversion

### PR DESCRIPTION
**Pull request by jonnyh404**
Moving some attributes to the base class for conversions.

### CODE REVIEW

**Review by cflynn3 — APPROVED** [review]
Looks good.

File: ResSimpy/Network.py [hunk 1]
```diff
@@ -6,15 +6,28 @@
 from ResSimpy.NodeConnections import NodeConnections
 from ResSimpy.Nodes import Nodes
 from ResSimpy.Targets import Targets
+from ResSimpy.WellConnections import WellConnections
+from ResSimpy.Wellbores import Wellbores
+from ResSimpy.Wellheads import Wellheads
 
 
 @dataclass(kw_only=True, init=False)
 class Network(ABC):
```
Comment (kwr3) [inline ResSimpy/Network.py:15]: consider doc string

File: ResSimpy/WellConnections.py [hunk 2]
```diff
@@ -1,14 +1,38 @@
...
+from typing import Literal, Optional, TYPE_CHECKING, Sequence
 
 from ResSimpy.OperationsMixin import NetworkOperationsMixIn
 from ResSimpy.WellConnection import WellConnection
+if TYPE_CHECKING:
+    from ResSimpy.Network import Network
 
 
+@dataclass(kw_only=True)
 class WellConnections(NetworkOperationsMixIn, ABC):
```
Comment (kwr3) [inline ResSimpy/WellConnections.py:13]: consider doc string for this class

Comment (kwr3) [inline ResSimpy/WellConnections.py:13]: is NetworkOperationsMixIn an ABC?

**Review by kwr3 — APPROVED** [review]
Looks good.

**Pull request merged**